aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -941,6 +947,72 @@ public void run() {
         }, "Mqtt Connect Thread").start();
     }
 
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using custom authorization tokens.
+     *
+     * @param tokenKeyName This specifies the key name that your device chooses, which indicates the
+     *                     token in the custom authorization HTTP request header. It is a string;
+     *                     the minimum length is 1 character, and the maximum length is 128 characters.
+     * @param token This specifies the custom authorization token to authorize the request to the AWS
+     *              IoT gateway. It is a string; the minimum length is 1 character, and the maximum
+     *              length is 1,024 characters.
+     * @param tokenSignature This is the token signature generated by using the private key with the
+     *                       SHA256withRSA algorithm.
+     * @param customAuthorizer Name of the AWS IoT custom authorizer.
+     * @param statusCallback When a new MQTT session status is received, a function of callback will
+     *                       be called with new connection status.
+     */
+    public void connect(
+            final String tokenKeyName,
+            final String token,
+            final String tokenSignature,
+            final String customAuthorizer,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (tokenKeyName == null || token == null || tokenSignature == null) {
+            throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
+        }
```

Comment:
This call seems to require API Level of 19 while the current minimum for IoT SDK is 11. We should probably increase the minimum API level but I feel 19 would too high given our user demographics.

Reply:
Ah, true. We can use it back to 15 in Amplify cause it gets "de-sugared". Anyway, sounds like its not worth the effort here since we use an old Java version and don't use retrolambda.